sycl-rfc-unnamed-lambda.md

```diff
@@ -0,0 +1,112 @@
+RFC: SYCL support for unnamed SYCL kernel functions
+
+The SYCL 2020 specification requires implementations to support the use of a lambda expression as
+the entry point to a SYCL kernel.
+Kernel invocation requires that host and device compilers agree on kernel names.
+Kernel names may be explicitly chosen by programmers by passing a type name as a template argument
+to a SYCL kernel invocation function like `sycl::handler::single_task`.
+In the following example, a kernel name is deterministically generated based on the `my_kernel_name`
+type.
+
+```c++
+#include <sycl/sycl.hpp>
+class my_kernel_name;
+int main() {
+  sycl::queue q(sycl::cpu_selector_v);
+  q.submit([](sycl::handler &cgh) {
+    cgh.single_task<my_kernel_name>([]{});
+  });
+  q.wait();
+}
+```
+
+If an explicit kernel name is not provided, then the implementation is required to implicitly
+generate a kernel name based on the kernel entry point and surrounding context.
+The above example can more simply be written as follows.
+Note that the declaration of `my_kernel_name` and all uses of it have been removed.
+
+```c++
+#include <sycl/sycl.hpp>
+int main() {
+  sycl::queue q(sycl::cpu_selector_v);
+  q.submit([](sycl::handler &cgh) {
+    cgh.single_task([]{});
+  });
+  q.wait();
+}
+```
+
+When the kernel entry point is a lambda expression, it is technically challenging for host
+and device compilers to independently generate a matching kernel name.
+The existing C++ name mangling schemes specify how names are generated for the closure type
+of a lambda expression, but these names are not generally intended to be stable; particularly
+not in the presence of conditionally included code.
+The SYCL 2020 specification therefore allows conforming implementations to support a reduced
+feature set that does not include support for unnamed SYCL kernel functions.
+This is specified in
+[Appendix B.2, "Reduced feature set"](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:feature-sets.reduced).
+Per
+[section 5.6, "Preprocessor directives and macros"](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_preprocessor_directives_and_macros),
+implementations that do not support unnamed SYCL kernel functions are required to predefine
+the `SYCL_FEATURE_SET_REDUCED` macro with the value `1` and to not define `SYCL_FEATURE_SET_FULL`.
+SYCL programs that use unnamed SYCL kernel functions are therefore more limited in their
+portability.
```

Comment:
Is all this information important for this document?  For example, do we need to talk about the details of the macros or even the fact that unnamed lambdas are only required for the full feature set?  It seems like this is all ancillary to the point you are trying to make.

Reply:
This is intended to explain why supporting unnamed lambdas when using a third party host compiler is technically challenging in order to provide justification for differentiating such support as presented in the proposal section.